src/main/java/com/autotune/common/datasource/DataSourceManager.java

```diff
@@ -0,0 +1,121 @@
+ * from collection of data sources
+ *
+ *
+ * Currently Supported Implementations:
+ *  - createExperimentsFromDataSource
+ */
+
+public class DataSourceManager {
+    private static final Logger LOGGER = LoggerFactory.getLogger(DataSourceManager.class);
+    DataSourceDetailsOperator dataSourceDetailsOperator = DataSourceDetailsOperator.getInstance();
+
+    public DataSourceManager() {
+    }
+
+    /**
+     * Imports Data for each data source using associated DataSourceInfo and DataSourceDetailsOperator.
+     * TODO - importDataFromAllDataSources is a temporary functionality added for demo purposes - to be deleted
+     */
+    public void importDataFromAllDataSources() {
+        try {
+            List<DataSourceDetailsInfo> dataSourceDetailsInfoList = new ArrayList<>();
+            HashMap<String, DataSourceInfo> dataSources = DataSourceCollection.getInstance().getDataSourcesCollection();
+            for (String name : dataSources.keySet()) {
+                DataSourceInfo dataSource = dataSources.get(name);
+                dataSourceDetailsOperator.createDataSourceDetails(dataSource);
+
+                DataSourceDetailsInfo dataSourceDetails = dataSourceDetailsOperator.getDataSourceDetailsInfo();
+                if (null == dataSourceDetails) {
+                    continue;
+                }
+                dataSourceDetailsInfoList.add(dataSourceDetails);
+            }
+            Gson gson = new GsonBuilder().setPrettyPrinting().create();
+            String jsonOutput = gson.toJson(dataSourceDetailsInfoList);
+
+            LOGGER.info(jsonOutput);
+        } catch (Exception e) {
+            LOGGER.error(e.getMessage());
+        }
+    }
+
+    /**
+     * Imports Data for a specific data source using associated DataSourceInfo.
+     */
+    public void importDataFromDataSource(DataSourceInfo dataSource) {
+
+        try {
+            if (null == dataSource) {
+                throw new DataSourceNotExist(KruizeConstants.DataSourceConstants.DataSourceErrorMsgs.MISSING_DATASOURCE_INFO);
+            }
+            dataSourceDetailsOperator.createDataSourceDetails(dataSource);
+        } catch (Exception e) {
+            LOGGER.error(e.getMessage());
+        }
+    }
+
+    /**
+     * Retrieves data from the specified data source information.
+     *
+     * @param dataSource The information about the data source to retrieve data from.
+     * @return DataSourceDetailsInfo containing details about the data source, or null if not found.
+     * @throws DataSourceNotExist Thrown when the provided data source information is null.
+     */
+    public DataSourceDetailsInfo getDataFromDataSource(DataSourceInfo dataSource) {
+
+        try {
+            if (null == dataSource) {
+                throw new DataSourceNotExist(KruizeConstants.DataSourceConstants.DataSourceErrorMsgs.MISSING_DATASOURCE_INFO);
+            }
+            return dataSourceDetailsOperator.getDataSourceDetailsInfo();
```

Comment:
This doesn't look right. The `getDataSourceDetailsInfo()` method should take in `DataSourceInfo` parameter and only return details about that `dataSource`

Reply:
I have updated the `getDataSourceDetailsInfo()` to return details specific to a given `dataSource`. The updated function returns a new instance of `DataSourceDetailsInfo` based on input dataSource provider as `DataSourceDetailsInfo` object can have metadata for multiple data sources/cluster groups.